text/0000-explicit-tail-calls.md

```diff
@@ -0,0 +1,775 @@
+- What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
+    - Are all calling-convention used by Rust available for TCE with the proposed restrictions on function signatures?
+    - Is there some way to reduce the impact on debugging and other features?
+- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of
+  the solution that comes out of this RFC?
+  - Supporting general tail calls, the current RFC restricts function signatures which can be loosened independently in the future.
+
+## Resolved Questions
+
+- Can generic functions be supported?
+  - As Rust uses Monomophization, generic functions are not a problem.
+- Can dynamic function calls be supported?
+  - Dynamic function calls are supported ([confirmation](https://github.com/rust-lang/rfcs/pull/3407#discussion_r1191600480)).
+- Can functions outside the current crate be supported, functions from dynamically loaded libraries?
+  - Same as dynamic function calls these function calls are supported ([confirmed for LLVM](https://github.com/rust-lang/rfcs/pull/3407#discussion_r1191602364)).
+- Can closures be supported?
+  - Closures are **not** supported see [here](#closures).
+- Can async functions be supported?
+  - Async functions are **not** supported see [here](#async).
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+<!-- Think about what the natural extension and evolution of your proposal would
+be and how it would affect the language and project as a whole in a holistic
+way. Try to use this section as a tool to more fully consider all possible
+interactions with the project and language in your proposal.
+Also consider how this all fits into the roadmap for the project
+and of the relevant sub-team.
+
+This is also a good place to "dump ideas", if they are out of scope for the
+RFC you are writing but otherwise related.
+
+If you have tried and cannot think of any future possibilities,
+you may simply state that you cannot think of anything.
+
+Note that having something written down in the future-possibilities section
+is not a reason to accept the current or a future RFC; such notes should be
+in the section on motivation or rationale in this or subsequent RFCs.
+The section merely provides additional information. -->
+
+## Helpers
+[helpers]: #helpers
+
+It seems possible to keep the restriction on exactly matching function signatures by offering some kind of placeholder
+arguments to pad out the differences. For example:
+
+```rust
+foo(a: u32, b: u32) {
+    // uses `a` and `b`
+}
+
+bar(a: u32, _b: u32) {
+    // only uses `a`
+}
+```
+
+Maybe it is useful to provide a macro or attribute that inserts missing arguments.
+
+```rust
+#[pad_args(foo)]
+bar(a: u32) {
+    // ...
+}
+```
```

Comment:
In those 2 examples `fn` tokens are missing

Reply:
whoops, updated: https://github.com/rust-lang/rfcs/pull/3407/commits/23106d2e678363ee4d4d6c6f16e535539dcd3e65